File: leetcode/230-kth-smallest-element-in-a-bst/kth-smallest-element-in-a-bst.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void Inorder(TreeNode* n,vector<int>& values){
        if(n==nullptr){
            return;
        }
        Inorder(n->left,values);
        values.push_back(n->val);
        Inorder(n->right,values);
    }
    int kthSmallest(TreeNode* root, int k) {
        vector<int> values;
        Inorder(root,values);
        return values[k-1];
    }
};

File: leetcode/230-kth-smallest-element-in-a-bst/kth-smallest-element-in-a-bst.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        result=[]
        def Inorder(node,k):
            if not node:
                return
            
            Inorder(node.left,k-1)
            result.append(node.val)
            Inorder(node.right,k-1)
        
        Inorder(root,k)
        return result[k-1]
        